internal/resourcemodifiers/resource_modifiers.go

```diff
@@ -95,13 +96,19 @@ func (p *ResourceModifiers) ApplyResourceModifierRules(obj *unstructured.Unstruc
 	return errs
 }
 
-func (r *ResourceModifierRule) Apply(obj *unstructured.Unstructured, groupResource string, log logrus.FieldLogger) error {
+func (r *ResourceModifierRule) apply(obj *unstructured.Unstructured, groupResource string, scheme *runtime.Scheme, log logrus.FieldLogger) error {
 	namespaceInclusion := collections.NewIncludesExcludes().Includes(r.Conditions.Namespaces...)
 	if !namespaceInclusion.ShouldInclude(obj.GetNamespace()) {
 		return nil
 	}
 
-	if r.Conditions.GroupResource != groupResource {
+	g, err := glob.Compile(r.Conditions.GroupResource)
```

Comment:
one thing came in my mind
if we support globbing with namespace filter.

Scenario -> user provided NS filter, with path glob as blah*
Now blah* can contain both cluster scope and namespaced scope CRs. 
We need to make it clear in docs that if user wants to do cluster scope they have to put NS filter as empty otherwise it won't work
and using it in conjunction with GroupResource glob path will have these further nuances.

Reply:
yeah
again the limitation is cluster scope won't be taken care of

Reply:
sorry, not quite understand, can you give an example?

Reply:
user provided NS filter (default), with **GroupResource** glob as blah*
Now blah* can contain both cluster scope and namespaced scope CRs.
We need to make it clear in docs that if user wants to do cluster scope they have to put NS filter as empty otherwise it won't work
and using it in conjunction with GroupResource glob path will have these further nuances.

Reworded my initial comment to make it more clear

Reply:
Hey I am not sure if ignoring ns filter is the right way
I am wondering if we should add another flag which says `includeClusterSCope` similar to what we have in backup/restore APIs.

Reply:
I think it's ok, ns filter is used to filter namespaced resources, cluster resources can not be filtered by ns, so we can just ignore it if both namespaced resources and cluster resources are included.

for example:
```
groupResource: *
namespaces:
  - ns1
```
this will match all namespaced resources in ns1 and all cluster resources

Reply:
but the issue is
we are doing cluster scope by default. Which is not the right behavior. 
We should give user all controls as possible. 


For now my suggestion would be revert the current chagne you did 

and behaviour should be as following

 -> NS field empty means both cluster scope and NS scoped will be changed
 
 NS field specified means only NS scoped. 
 
 IF a use wishes to do both cluster scope and specific NS scope they can use multiple resource modifier rules
 
 For Example -> The first rule can be
 GroupResource - provider.clusterscoperesources.*
 namespace 
- ns1

Second rule

GroupResource - provider.*
namespace (blank)


My request would be

1. to document this behaviour of NS field empty = cluster scope all + all NS 
And revert the change you have done for now - which always includes cluster scope.

if this suggestion doesn't make sense 

2. the altnerative would be to introduce new clsuterSCope boolean field for clarity 


i would suggest to go with approach 1 for now.

Reply:
if the user needs multiple rm rules to do one thing, it is too redundant.
Maybe we need a REAL resource fileters here, to support all conditions (for example namespaced resource a1 of A + cluster resource b2 of B ), but it is too complex and will break the compatibility.
My idea is to keep it simple, I think including the cluster resources by default is reasonable

Reply:
" I think including the cluster resources by default is reasonable"
I feel this would lead to changes which are not intended.
I am worried about this.

In terms of REAL filters here - similar to what we have in backup /restore - includeClusterSCopeResources
We can add a bool for that. 
that would make it consistent and give control to user.

Reply:
This is only useful for globbed patterns
but would lead to confusion if explicit GroupResource is mentioned.

Reply:
On further thought, I think it's fine.
But PLEASE just add this behaviour of cluster scope by default to docuemtnation.

Reply:
Just to reiterate, I am fine with current changes you have. Just that I request you to document the behaviour of cluster scope by default in case of globbed paths.

Reply:
thanks, updated